src/main/java/dev/slimevr/vr/processor/skeleton/SimpleSkeleton.java

```diff
@@ -680,21 +692,20 @@ protected void updateLocalTransforms() {
 
 		// Spine
 		if (hasSpineTracker) {
-			if (chestTracker.getRotation(rotBuf1)) {
-				neckNode.localTransform.setRotation(rotBuf1);
-			}
-			if (waistTracker.getRotation(rotBuf1)) {
-				chestNode.localTransform.setRotation(rotBuf1);
-				trackerChestNode.localTransform.setRotation(rotBuf1);
-			}
-			if (hipTracker.getRotation(rotBuf1)) {
-				waistNode.localTransform.setRotation(rotBuf1);
-				hipNode.localTransform.setRotation(rotBuf1);
-				trackerWaistNode.localTransform.setRotation(rotBuf1);
-			}
-		} else if (hmdTracker != null) { // If no spine tracker, allign spine
-											// yaw with HMD
-			rotBuf1 = rotBuf1.fromAngles(0, rotBuf1.getYaw(), 0);
+			getFirstAvailableTracker(chestTracker, waistTracker, hipTracker).getRotation(rotBuf1);
+			neckNode.localTransform.setRotation(rotBuf1);
+
+			getFirstAvailableTracker(waistTracker, hipTracker, chestTracker).getRotation(rotBuf1);
+			chestNode.localTransform.setRotation(rotBuf1);
+			trackerChestNode.localTransform.setRotation(rotBuf1);
+
+			getFirstAvailableTracker(hipTracker, waistTracker, chestTracker).getRotation(rotBuf1);
+			waistNode.localTransform.setRotation(rotBuf1);
+			hipNode.localTransform.setRotation(rotBuf1);
+			trackerWaistNode.localTransform.setRotation(rotBuf1);
+		} else if (hmdTracker != null) {
+			// Allign spine yaw with HMD
```

Comment:
Spelling mistake in "Align"
```suggestion
			// Align spine yaw with HMD
```

Reply:
I hate how I saw that and fixed it in another branch where I was testing stuff ;-;